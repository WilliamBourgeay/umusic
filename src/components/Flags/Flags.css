.arrow-down {
  color: transparent !important;
}

.flag-select .arrow-down {
  display: none !important;
}
