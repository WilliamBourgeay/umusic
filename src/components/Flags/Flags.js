// @flow

import React from 'react';
import ReactFlagsSelect from 'react-flags-select';

import 'react-flags-select/css/react-flags-select.css';
import './Flags.css';

type Props = {
  countries: string[],
  defaultCountry: string,
  onSelect: Function,
  customLabels: Object,
};

const Flags = ({ countries, defaultCountry, onSelect, customLabels }: Props) => (
  <ReactFlagsSelect
    className="arrow-down"
    countries={countries}
    selectedSize={20}
    defaultCountry={defaultCountry}
    customLabels={customLabels}
    showOptionLabel={false}
    showSelectedLabel={false}
    onSelect={onSelect}
  />
);

export default Flags;
