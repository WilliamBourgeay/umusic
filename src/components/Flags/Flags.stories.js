import React from 'react';
import { storiesOf } from '@storybook/react';
import Flags from './Flags';

const stories = storiesOf('Flags', module);

stories.add('default', () => (
  <Flags countries={['GB', 'FR']} defaultCountry="FR" onSelect={() => {}} />
));
