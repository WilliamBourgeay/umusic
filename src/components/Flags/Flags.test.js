import * as React from 'react';
import renderer from 'react-test-renderer';

import Flags from './Flags';

describe('Flags', () => {
  const f = () => {};

  it('should render a Flags', () => {
    const tree = renderer.create(
      <Flags
        countries={['FR', 'DE']}
        defaultCountry="FR"
        onSelect={f}
        customLabels={{ FR: 'fr', DE: 'de' }}
      />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
