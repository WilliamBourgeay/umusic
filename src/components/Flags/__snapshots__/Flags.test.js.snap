// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flags should render a Flags 1`] = `
<div
  className="flag-select arrow-down"
>
  <div
    className="selected--flag--option "
    onClick={[Function]}
    onKeyUp={[Function]}
    style={
      Object {
        "fontSize": "20px",
      }
    }
    tabIndex="0"
  >
    <span
      className="country-flag"
      style={
        Object {
          "height": "20px",
          "width": "20px",
        }
      }
    >
      <img
        src="fr.svg"
      />
    </span>
    <span
      className="arrow-down "
    >
      â–¾
    </span>
  </div>
</div>
`;
